compiler/backend/src/main/java/org/apache/nemo/compiler/backend/nemo/NemoPlanRewriter.java

```diff
@@ -101,34 +122,41 @@ public PhysicalPlan rewrite(final PhysicalPlan currentPhysicalPlan, final int me
     // Optimize using the Message
     final Message message = new Message(messageId, examiningEdges, aggregatedData);
     final IRDAG newIRDAG = nemoOptimizer.optimizeAtRunTime(currentIRDAG, message);
+    this.setCurrentIRDAG(newIRDAG);
 
     // Re-compile the IRDAG into a physical plan
     final PhysicalPlan newPhysicalPlan = nemoBackend.compile(newIRDAG);
 
     // Update the physical plan and return
     final List<Stage> currentStages = currentPhysicalPlan.getStageDAG().getTopologicalSort();
     final List<Stage> newStages = newPhysicalPlan.getStageDAG().getTopologicalSort();
-    for (int i = 0; i < currentStages.size(); i++) {
+    IntStream.range(0, currentStages.size()).forEachOrdered(i -> {
       final ExecutionPropertyMap<VertexExecutionProperty> newProperties = newStages.get(i).getExecutionProperties();
       currentStages.get(i).setExecutionProperties(newProperties);
-    }
+      newProperties.get(ParallelismProperty.class).ifPresent(newParallelism -> {
+        currentStages.get(i).getTaskIndices().clear();
+        currentStages.get(i).getTaskIndices().addAll(IntStream.range(0, newParallelism).boxed()
+          .collect(Collectors.toList()));
+        IntStream.range(currentStages.get(i).getVertexIdToReadables().size(), newParallelism).forEach(newIdx ->
+          currentStages.get(i).getVertexIdToReadables().add(new HashMap<>()));
+      });
+    });
     return currentPhysicalPlan;
   }
 
   @Override
-  public void accumulate(final int messageId, final Object data) {
+  public void accumulate(final int messageId, final Set<StageEdge> targetEdges, final Object data) {
+    final Prophet prophet;
+    final List<ControlMessage.RunTimePassMessageEntry> parsedData = (List<ControlMessage.RunTimePassMessageEntry>) data;
+    if (!parsedData.isEmpty() && parsedData.get(0).getKey().equals("NONE")) {
```

Comment:
Can you leave a short comment on this magic number 0?

Reply:
It would be better if "NONE" is a public/global/static variable, rather than a local string